[GA] Add Irish language support for HassGetCurrentTime intent

GGamerzz opened the pull request:
- Implemented intent sentences in Irish for querying the current time
- Added templated response to generate natural time phrases in Irish
- Ensured compatibility with live slotless queries (e.g., “cén t-am é”)
- Included unit tests for common time-related phrases

AlanGreene reviewed (commented):
Mise arís (it's me again…) with a few questions 👋

I almost missed this as it doesn't have the `lang: ga` label 🤷

GGamerzz commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,91 @@ language: ga
+        } %}
 
-        "
+        {% set minutes_map = {
```
I think this is the biggest hurdle/challenge we will come across, asking my family we agree that adding `nóiméad` into every case would be better, personally I find it like its taking a shortcut when you leave it out. 
So I think I will leave the `nóiméad` in.

AlanGreene replied:
Fair enough. I suspect it may be a regional / dialect thing anyway (I have fairly mixed Irish…), so at least being self-consistent is good 👍

AlanGreene replied:
Just a point of reference: https://www.focloir.ie/en/dictionary/ei/one#one__4:~:text=twenty%20to%20one-,fiche%20chun%20a%20haon,-%2C%20fiche%20go

Foras na Gaeilge omit `nóiméad` in the examples given.

GGamerzz commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,91 @@ language: ga
+          18: "ocht nóiméad déag",
+          19: "naoi nóiméad déag",
+          20: "fiche nóiméad",
+          21: "fiche nóiméad agus nóiméad amháin",
```
`20: "fiche nóiméad",
 21: "fiche a haon nóiméad",
 22: "fiche a dó nóiméad",   
    23: "fiche a trí nóiméad",  
          24: "fiche a ceithre nóiméad",  
          25: "fiche a cúig nóiméad",     
          26: "fiche a sé nóiméad",       
          27: "fiche a seacht nóiméad",   
          28: "fiche a hocht nóiméad",    
          29: "fiche a naoi nóiméad"    

This would be more normal to me and how I would say it, adding the "a" in the middle is the issue, i wouldn't say "fiche a dó", I would say "fiche dó". Let me know what you think of that

AlanGreene replied:
I would be the same and not actually say the `a`, I think it's just a matter of pronunciation / spoken vs. written form, since the `fichE A …` is essentially combined…

Shouldn't `fiche a ceithre` be `fiche a ceathair` though?

I also wonder how much dialect plays into this, I opened a related discussion last night and would be interested in your thoughts: https://github.com/OHF-Voice/intents/discussions/3238

GGamerzz commented:
I am going to update my current code so far, what i've changed and updated for you to have a look, any other issues that are outstanding or come up, we can work on that too

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,91 @@ language: ga
     HassGetCurrentTime:
-      default: "TODO: {% set minute_str = '{0:02d}'.format(slots.time.minute) %}
+      default: >
+        {% if slots.time is defined %}
```
Yeah I'd use `tráthnóna` to cover the whole period between noon and night. Rarely if ever use `iarnóin`.

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,105 @@ language: ga
+          1: "nóiméad amháin",
+          2: "dhá nóiméad",
+          3: "trí nóiméad",
+          4: "ceathair nóiméad",
```
```suggestion
          4: "ceithre nóiméad",
```

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,105 @@ language: ga
+          11: "aon nóiméad déag",
+          12: "dhá nóiméad déag",
+          13: "trí nóiméad déag",
+          14: "ceathair nóiméad déag",
```
```suggestion
          14: "ceithre nóiméad déag",
```

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,105 @@ language: ga
     HassGetCurrentTime:
-      default: "TODO: {% set minute_str = '{0:02d}'.format(slots.time.minute) %}
+      default: >
+        {% if slots.time is defined %}
+          {% set hour = slots.time.hour %}
+          {% set minute = slots.time.minute %}
+        {% elif context['time.hour'] is defined %}
+          {% set hour = context['time.hour'] | int %}
+          {% set minute = context['time.minute'] | int %}
```
Looking at the files for other languages it appears safe to assume that both `slots.time.hour` and `slots.time.minute` are defined. This can be reduced to just declaring the `hour` and `minutes` variables directly unless there was a particular reason for the `context` fallback here.

```suggestion
        {% set hour = slots.time.hour %}
        {% set minute = slots.time.minute %}
```

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,105 @@ language: ga
+        {% elif context['time.hour'] is defined %}
+          {% set hour = context['time.hour'] | int %}
+          {% set minute = context['time.minute'] | int %}
+        {% else %}
+          {% set hour = 1 %}
+          {% set minute = 2 %}
```
I'm not sure I understand what this is for. It seems odd to set an arbitrary time, and given my above comment about the hour and minute appearing to always be defined this shouldn't be necessary.

GGamerzz replied:
I believe this was me trying to get the tests working and I should've removed before pull request

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,105 @@ language: ga
+          {% set minute = 2 %}
+        {% endif %}
+        {% set next_hour = (hour + 1) % 24 %}
+        {% set is_default = (hour == 1 and minute == 2) %}
```
same as above, not sure why this is needed

GGamerzz replied:
same as above for me, shoulda been removed after testing

AlanGreene reviewed (changes requested):
Hey @GGamerzz, not sure if this is ready for another review but I left a few comments / questions anyway. Apologies if I jumped the gun.

GGamerzz commented:
Not sure why it is not building, will have a look during the week, if you spot the issue let me know and ill get to it

AlanGreene commented on responses/ga/HassGetCurrentTime.yaml:
```diff
@@ -2,11 +2,95 @@ language: ga
+        } %}
+
+        {% set time_suffix = "" %}
+        {% if is_default %}
+           {% set time_suffix = " san oíche" %}
```
This needs to be removed too
